### ISSUE TITLE
Option to have absolute URLs in playlists and serve the playlist from the server, not storage

### ISSUE THREAD

**Issue by notnownikki** [owncast/owncast#1368]
We stream to a fair few folk around the world, so we're using a CDN to distribute the video segments, and the playlists also end up there.

The problem is, the CDN we're using is rate limiting us because we're pushing updates to the playlists every few seconds.

I'd love on option to be able to serve the playlists from the server, and have full URLs in the playlists so that the segments get served from the CDN, but the playlist gets served from the server.

I thought that if a custom CDN prefix is specified (which currently says that you probably don't need it :) ) then there could be an option to make the playlists work this way. It would mean our friends in AUS would be able to get the segments nice and fast, and only have to hit our server in Europe for playlist updates.

**Comment by gabek** [comment]
With a CDN you wouldn't push files to it, so I'm a little confused of your exact situation.
 
But regardless how your CDN is working, it seems like a significant limitation of the CDN.  If you can't distribute the playlists through the CDN that's half of the puzzle.  Those need to get to the users quickly, too.  I understand they're small, but breaking up the variant playlists and the segments that go with them feel weird.  In fact, it used to work that way with Owncast, and I fixed it to work the way it does now since if a stream has significant load and a lot of viewers, them hammering the single Owncast server for playlist updates will take a toll.

To be honest I'm not 100% clear what your setup is, or what's being used, but it sounds like you need a different CDN if it's somehow limiting what it's proxying.  Feel free to clarify.

**Comment by notnownikki** [comment]
Sorry yes I got my terms completely mixed up, been looking at this so long that I didn't make any sense :) The issue has nothing to do with the CDN, thanks for responding to my absolute nonsense 🤦‍♀️

So I use the digital ocean spaces object storage, and in the documentation at https://docs.digitalocean.com/products/spaces/ it says we're limited to 5 PUTs on a single object every 5 minutes, so updating the playlist as often as we do for a livestream is way over that limit, and I'm trying to look for a solution so that DO don't end up suspending my access.

<blockquote><img src="https://assets.digitalocean.com/articles/pdocs/docs-banner.png" width="48" align="right"><div>DigitalOcean</div><div><strong><a href="https://docs.digitalocean.com/products/spaces/">Spaces | DigitalOcean Documentation</a></strong></div><div>Spaces is an S3-compatible object storage service that lets you store and serve large amounts of data. The free, built-in Spaces CDN minimizes page load times, improves performance, and reduces bandwidth and infrastructure costs.</div></blockquote>

**Comment by gabek** [comment]
Oh, got it!

I didn't realize DigitalOcean had those restrictions. I've used DigitalOcean myself and not run into any issues.  I have DigitalOcean Spaces + their CDN enabled on https://nightly.owncast.tv and haven't had problems, but that doesn't mean they couldn't pull the trigger.

I think the answer is simply if DigitalOcean isn't compatible with what you or I are doing we'd just have to stop using them and use some other company.  While we could add a bunch of different switches and micromanagement options for certain files to work in different ways it would add complexity to both the configuration of things and to the codebase just to work around a company who doesn't want the business.

So in short, it's worked ok for me.  In fact I'm watching DigitalOcean Spaces served content right now from an Owncast server, but it's up to you if that risk is acceptable.

I know it's not a satisfactory answer for your specific use case, but in general I don't want to add overarching complexity as very specific workarounds.

<blockquote><div><strong><a href="https://nightly.owncast.tv/">Owncast</a></strong></div></blockquote>

**Comment by notnownikki** [comment]
> I know it's not a satisfactory answer for your specific use case, but in general I don't want to add overarching complexity as very specific workarounds.

No, that's an absolutely valid answer! Thank you :)

I've implemented the workaround myself for now, until I find somewhere with object storage and a CDN that's as affordable as DO.
